libs/components/lookup/src/lib/modules/lookup/lookup.component.spec.ts

```diff
@@ -3378,6 +3456,143 @@ describe('Lookup component', function () {
+                'searchFunctionFilterModal',
+                friends
+              );
+
+              component.customSearchFilters = [searchFilterFunctionSpy];
+              fixture.detectChanges();
+              tick();
+
+              // opening the "show more" modal
+              component.enableShowMore = true;
+
+              fixture.detectChanges();
+              tick();
+
+              performSearch('s', fixture);
+              clickShowMore(fixture);
+
+              // peforming a search in the modal view should provide the search filter function with a 'modal' context
+              performSearch('s', fixture);
+              expect(searchFilterFunctionSpy).toHaveBeenCalledWith(
+                's',
+                jasmine.objectContaining({
+                  name: jasmine.any(String),
+                }),
+                jasmine.objectContaining({
+                  context: 'modal',
+                })
+              );
+            }));
+          });
+        });
+        describe('custom search function', () => {
+          let customSearchFunctionSpy: jasmine.Spy;
+
+          it('should return popover context', fakeAsync(() => {
+            // creating the spy function that is pretending to be a custom search function with real functionality, because a real function must be provided in order to bypass an error with undefined results returned (the Spy, though not a function, is not undefined so the Lookup/Autocomplete component actually tries to search using it)
+            customSearchFunctionSpy = getCustomSearchFunctionSpy(
+              'customSearchFunctionPopover',
+              friends
+            );
+
+            component.customSearch = customSearchFunctionSpy;
+            // peforming a search in the popover view should provide the custom search function with a 'popover' context
+            fixture.detectChanges();
+            tick();
+
+            performSearch('s', fixture);
+            expect(customSearchFunctionSpy).toHaveBeenCalledWith(
+              's',
+              jasmine.arrayWithExactContents(component.data),
+              jasmine.objectContaining({
+                context: 'popover',
+              })
+            );
+          }));
+
+          describe('show more modal', () => {
+            // This is necessary as due to modals being launched outside of the test bed they will not
+            // automatically be disposed between tests.
+            it('should return modal context', fakeAsync(() => {
+              fixture.detectChanges();
+              tick();
+              const friends: any[] = component.form.controls.friends.value;
+              // creating the spy function that is pretending to be a custom search function with real functionality, because a real function is needed to be called so that some results may appear in the popover/dropdown so that the "Show more" button may be present so it can be clicked
```

Comment:
Last thing: let's move this duplicated comment above the `getCustomSearchFunctionSpy` function (here, and a few lines above), using JSDocs format.

Reply:
done in [clean up comments, spelling, & spacing](https://github.com/blackbaud/skyux/pull/381/commits/8bdc105bba11d296931923f4eea9434e7a36549d)